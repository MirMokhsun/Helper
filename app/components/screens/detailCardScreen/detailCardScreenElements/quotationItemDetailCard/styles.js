import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        height: 25,
        justifyContent: 'center',
    },
    whiteNumbers: {
        lineHeight: 25,
        fontSize: 20,
        color: "#9FA0B2",
        fontFamily: 'Roboto-Regular',
    },
    largeNumbers: {
        paddingTop: 0.5,
        fontSize: 23,
    },
    valuestyle: {

    },
    redNumbers: {
        lineHeight: 25,
        fontSize: 20,
        color: "#F86F6F",
        fontFamily: 'Roboto-Regular',
    },
    greenNumbers: {
        lineHeight: 25,
        fontSize: 20,
        color: "#1459D2",
        fontFamily: 'Roboto-Regular',
    },
})