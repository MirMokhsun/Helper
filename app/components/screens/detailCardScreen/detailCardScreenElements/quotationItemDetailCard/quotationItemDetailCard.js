import React from 'react';
import { View, Text, I18nManager } from 'react-native';
import { styles } from './styles';
import { isEqualNumb, setCamelPos } from '../../../../../api/cardQuotation.api/cardquotation.api';

export class QuotationItemDetailCard extends React.PureComponent {
    constructor(props) {
        super(props)

        this.countUperNambers = 2;
        this.rendersNumb = setCamelPos(props.number, styles.whiteNumbers, this.countUperNambers, styles);
        this.rendersNumbWithRtl = I18nManager.isRTL ? this.rendersNumb.reverse() : this.rendersNumb;
        this.state = {
            rendersNumb: this.rendersNumbWithRtl,
        }
    }

    componentWillReceiveProps({ number }) {
        const { number: thisNumber } = this.props;
        const compare = isEqualNumb(thisNumber, number);
        const colorNumbers = compare ? styles.greenNumbers : styles.redNumbers;
        let rendersNumb = setCamelPos(number, colorNumbers, this.countUperNambers, styles);
        rendersNumb = I18nManager.isRTL ? rendersNumb.reverse() : rendersNumb;

        this.setState({
            rendersNumb,
        });
    }


    render() {
        const { rendersNumb } = this.state;

        return (
            <View style={styles.container}>
                {rendersNumb}
            </View >
        )
    }
}

export default QuotationItemDetailCard;