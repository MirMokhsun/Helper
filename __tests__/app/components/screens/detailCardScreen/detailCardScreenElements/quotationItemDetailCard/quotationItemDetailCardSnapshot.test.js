import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { QuotationItemDetailCard } from '../../../../../../../app/components/screens/detailCardScreen/detailCardScreenElements/quotationItemDetailCard/quotationItemDetailCard';

it('create snapshot PivotPointAndChartsLine', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<QuotationItemDetailCard number={'0.987465'} />    );
    expect(toolbarSnapshot).toMatchSnapshot();
});