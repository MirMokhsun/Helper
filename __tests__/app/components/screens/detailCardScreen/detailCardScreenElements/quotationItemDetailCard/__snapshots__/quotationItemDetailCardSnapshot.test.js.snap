// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot PivotPointAndChartsLine 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 25,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#9FA0B2",
          "fontFamily": "Roboto-Regular",
          "fontSize": 20,
          "lineHeight": 25,
        },
        Object {},
      ]
    }
  >
    0
    .
    9
    8
    7
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#9FA0B2",
          "fontFamily": "Roboto-Regular",
          "fontSize": 20,
          "lineHeight": 25,
        },
        Object {
          "fontSize": 23,
          "paddingTop": 0.5,
        },
      ]
    }
  >
    4
    6
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#9FA0B2",
          "fontFamily": "Roboto-Regular",
          "fontSize": 20,
          "lineHeight": 25,
        },
        Object {},
      ]
    }
  >
    5
  </Text>
</View>
`;
